### PR TITLE
Adding tests for inheritance

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Adding tests for inheritance

**Related issue(s)**:
[[Solidity Support] Inheritance operations#446](https://github.com/hashgraph/hedera-smart-contracts/issues/446)

### CODE REVIEW

File: test/solidity/inheritance/inheritance.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const TOP_UP_AMOUNT = ethers.utils.parseEther('0.000001');
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    wallet = signers[0];
+
+    const factoryMain = await ethers.getContractFactory('Main')
+    contractMain = await factoryMain.deploy()
+    await contractMain.deployed()
+    contractMainAddr = contractMain.address
```
Comment (natanasow) [inline test/solidity/inheritance/inheritance.js:15]: we can remove the redundant variable `contractMainAddr` if it's used only once in `const tx = {}` and use directly `contractMain.address`

File: test/solidity/inheritance/inheritance.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    const factoryBase = await ethers.getContractFactory('Base')
+    contractBase = await factoryBase.deploy()
+    await contractBase.deployed()
+    contractBaseAddr = contractBase.address
```
Comment (natanasow) [inline test/solidity/inheritance/inheritance.js:20]: same redundant variable (`contractBaseAddr`)

File: test/solidity/inheritance/inheritance.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    expect(res).to.equal("Main")
+  })
+
+  it('selfdestruct(address payable recipient)', async function () {
+    const balanceBaseInitial = await contractBase.getBalance()
+    expect(balanceBaseInitial).to.be.equal(0)
+
+    const tx = await contractMain.destroyContract(contractBaseAddr)
+    tx.wait()
```
Comment (natanasow) [inline test/solidity/inheritance/inheritance.js:48]: the promise should be awaited `await tx.wait()`

File: contracts/solidity/inhetitance/Main.sol [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    function returnSuper() public view virtual returns (string memory) {
+        return super.classIdentifier();
+    }
+
+    function destroyContract(address recipient) public returns (bool) {
+        selfdestruct(payable(recipient));
+
+        return true;
```
Comment (natanasow) [inline contracts/solidity/inhetitance/Main.sol:21]: we can remove the return statement if it's not used, the test should check for balance before/after `selfdestruct` as it is now

**Review by natanasow — CHANGES_REQUESTED** [review]
LGTM, some nits

File: contracts/solidity/inhetitance/Base.sol [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+contract Base {
+    receive() external payable {}
+    function classIdentifier() public pure virtual returns (string memory) {
+        return "Base";
+    }
+
+    function getBalance() public view returns (uint256) {
+        return address(this).balance;
+    }
+}
```
Comment (natanasow) [inline contracts/solidity/inhetitance/Base.sol:13]: new empty line must be added

File: contracts/solidity/inhetitance/Main.sol [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    }
+}
```
Comment (natanasow) [inline contracts/solidity/inhetitance/Main.sol:23]: new empty line must be added

File: test/solidity/inheritance/inheritance.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    //top up the test contract with some funds
+    const tx = {
+        to: contractMainAddr,
+        value: TOP_UP_AMOUNT
+    }
+    const topUpRes = await wallet.sendTransaction(tx)
+    await topUpRes.wait();
+  })
+
+  it("it should confirm solidity functionality: this (current contract's type)", async function () {
```
Comment (konstantinabl) [inline test/solidity/inheritance/inheritance.js:31]: use single quotes and remove the 'it' from the description everywhere

File: test/solidity/inheritance/inheritance.js (hunk 1 above)
Comment (stefan-stefanooov) [reply test/solidity/inheritance/inheritance.js:15]: It's used on line 34 as well

**Review by konstantinabl — DISMISSED** [review]

**Comment by stefan-stefanooov** [comment]
Nothing changed. Fixing merge conflicts.

**Review by natanasow — APPROVED** [review]

**Pull request merged**